x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativeRolesStore.java

```diff
@@ -195,66 +195,59 @@ public void onFailure(Exception e) {
-                executeAsyncWithOrigin(client.threadPool().getThreadContext(), SECURITY_ORIGIN,
-                        client.prepareMultiSearch()
-                                .add(client.prepareSearch(SecurityIndexManager.SECURITY_INDEX_NAME)
-                                        .setQuery(QueryBuilders.termQuery(RoleDescriptor.Fields.TYPE.getPreferredName(), ROLE_TYPE))
-                                        .setSize(0))
-                                .add(client.prepareSearch(SecurityIndexManager.SECURITY_INDEX_NAME)
-                                        .setQuery(QueryBuilders.boolQuery()
-                                                .must(QueryBuilders.termQuery(RoleDescriptor.Fields.TYPE.getPreferredName(), ROLE_TYPE))
-                                                .must(QueryBuilders.boolQuery()
-                                                        .should(existsQuery("indices.field_security.grant"))
-                                                        .should(existsQuery("indices.field_security.except"))
-                                                        // for backwardscompat with 2.x
-                                                        .should(existsQuery("indices.fields"))))
-                                        .setSize(0)
-                                        .setTerminateAfter(1))
-                                .add(client.prepareSearch(SecurityIndexManager.SECURITY_INDEX_NAME)
-                                        .setQuery(QueryBuilders.boolQuery()
-                                                .must(QueryBuilders.termQuery(RoleDescriptor.Fields.TYPE.getPreferredName(), ROLE_TYPE))
-                                                .filter(existsQuery("indices.query")))
-                                        .setSize(0)
-                                        .setTerminateAfter(1))
-                                .request(),
-                        new ActionListener<MultiSearchResponse>() {
-                            @Override
-                            public void onResponse(MultiSearchResponse items) {
-                                Item[] responses = items.getResponses();
-                                if (responses[0].isFailure()) {
-                                    usageStats.put("size", 0);
-                                } else {
-                                    usageStats.put("size", responses[0].getResponse().getHits().getTotalHits());
-                                }
-
-                                if (responses[1].isFailure()) {
-                                    usageStats.put("fls", false);
-                                } else {
-                                    usageStats.put("fls", responses[1].getResponse().getHits().getTotalHits() > 0L);
-                                }
+        Map<String, Object> usageStats = new HashMap<>(3);
+        securityIndex.prepareIndexIfNeededThenExecute(listener::onFailure, () ->
```

Comment:
Doesn't this mean that calling usage stats will trigger the creation of the secuity index?
I don't necessarily have an objection, but it doesn't seems like something we want to just slip into this PR.